src/sparseml/optim/helpers.py

```diff
@@ -195,6 +195,68 @@ def parse_recipe_variables(
     return recipe_variables
 
 
+def _update_staged_container_value(var_key, var_value, container):
+    """
+    Extends the functionality of _update_container_value to handle staged containers.
+
+    :param var_key: The key we are attempting to find in the container.
+    :param var_value: The value which will overwrite the previous value
+        every time var_key is found in the container's keys.
+    :param container: A container generated from a YAML string of SparseML recipe
+    :return: (optionally mutated) container, as well as key_value
+        (True if var_key found in container's attributes, otherwise False)
+    """
+    key_found = False
+
+    for container_key, container_value in container.items():
+        if not isinstance(container_value, dict):
+            # checking contents of global variables
+            if var_key == container_key:
+                container[var_key] = var_value
+                key_found = True
+
+        else:
+            # checking contents of a stage
+            stage_container, stage_key_found = _update_container_value(
+                var_key, var_value, container_value
+            )
+            container[container_key] = stage_container
+            if stage_key_found:
+                key_found = True
+
+    return container, key_found
+
+
+def _update_container_value(var_key: str, var_value, container):
```

Comment:
let's rename `_update_container_value ` to `_update_recipe_variable` as container is a little ambiguous outside of the scope of the functions